pkg/collector/process.go

```diff
@@ -209,7 +214,7 @@ func (cp *CollectingProcess) decodeTemplateSet(templateBuffer *bytes.Buffer, obs
 	if err := templateSet.PrepareSet(entities.Template, templateID); err != nil {
 		return nil, err
 	}
-	elementsWithValue := make([]*entities.InfoElementWithValue, int(fieldCount))
+	elementsWithValue := make([]entities.InfoElementWithValue, int(fieldCount))
```

Comment:
Did this change make any difference? This variable elementsWithValue's lifecycle does not seem to go beyond its enclosing function.

Reply:
Yes, it did. Hypothesis is allocation pattern. With change, we allocate contiguous memory at once, whereas with pointer we could allocate the memory anywhere. The latter causes memory defragmentation.

Reply:
This means previously, the InfoElementWithValue objects are allocated on the heap?

Reply:
Could there be a minor optimization to use an array instead of slice, to save 12 bytes and one pointer created for the slice itself? Not important right now, perhaps something we can try later on.

Reply:
> Could there be a minor optimization to use an array instead of slice, to save 12 bytes and one pointer created for the slice itself? Not important right now, perhaps something we can try later on.

It's possible. However, I am suspecting the array may not perform well when we try to pass by value. There is a callee that takes in by value. We can move that to pointer and see the diff.

Reply:
👍